test/functional/test_framework/socks5.py

```diff
@@ -117,6 +128,33 @@ def handle(self):
             cmdin = Socks5Command(cmd, atyp, addr, port, username, password)
             self.serv.queue.put(cmdin)
             logger.debug('Proxy: %s', cmdin)
+
```

Comment:
2b2e747455699236200fa3cddddfff7005f6a44a Could be overkill, but you could actually adjust the "dummy response" a few lines up to actually respond accurately, in particular the `REPLY` field could be used to report a connection failure back to bitcoind.

From https://www.rfc-editor.org/rfc/rfc1928

```diff
diff --git a/test/functional/test_framework/socks5.py b/test/functional/test_framework/socks5.py
index 48ad8bbeea..568e9fa64c 100644
--- a/test/functional/test_framework/socks5.py
+++ b/test/functional/test_framework/socks5.py
@@ -123,7 +123,14 @@ class Socks5Connection():
             port = (port_hi << 8) | port_lo
 
             # Send dummy response
-            self.conn.sendall(bytearray([0x05, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]))
+            self.conn.sendall(bytearray([
+                0x05, # Protocol version
+                0x00, # Reply (0x00 = success)
+                0x00, # Reserved (must be 0x00)
+                0x01, # Server bound address type (0x01 = ipv4)
+                0x00, 0x00, 0x00, 0x00, # Server bound ipv4 address
+                0x00, 0x00 # Server bound port
+            ]))
 
             cmdin = Socks5Command(cmd, atyp, addr, port, username, password)
             self.serv.queue.put(cmdin)

```

Reply:
I will leave it as it is, looks too complicated after looking at doing this for some time.